service/src/main/java/com/codeforcommunity/dataaccess/AuthDatabaseOperations.java

```diff
@@ -90,4 +93,47 @@ public boolean isOnBlackList(String signature) {
             Tables.BLACKLISTED_REFRESHES
                 .where(Tables.BLACKLISTED_REFRESHES.REFRESH_HASH.eq(signature)));
     }
+
+    public void createSecretKey(int userId, String token) throws AuthException {
+        if (!db.fetchExists(Tables.NOTE_USER.where(NOTE_USER.ID.eq(userId)))) {
+            throw new AuthException("User does not exist.");
```

Comment:
With the changes to how we're doing exception handling then this will be replaced with a custom exception.